R/utils.R

```diff
@@ -598,3 +598,46 @@ wait_for_nonmem.list <- function(.mod, .time_limit = 300, .interval = 5) {
 
 }
 
+
+#' Replace BBI_NULL_NUM and BBI_NULL_STR with NA_real_ or NA_character_
+#'
+#' @param .vals a numeric or character vector
+#'
+#' @importFrom purrr map
+#' @keywords internal
+set_bbi_null <- function(.vals){
+  map(.vals, function(.val){
+    if(inherits(.val, c("integer", "numeric", "character")) && (.val == BBI_NULL_NUM)){
+      return(NA_real_)
+    }else{
+      return(.val)
+    }
+  }) %>% unlist()
```

Comment:
Seems worth considering whether we're happy to just go with `NA` and just let type coercion do it's thing.  In that case, this whole function could be reduced to two lines:

```
  is.na(.vals[.vals == BBI_NULL_NUM]) <- TRUE
  .vals
```

Depending on the other values in the vector, that could lead to an `NA_integer_` when a `NA_real_` would be more appropriate, but I don't think that's a problem and the simplicity is nice.  (You could of course also keep the vector subsetting approach and add more conditions to handle the different types.)

Reply:
Im going to keep the functions the way they are for now, as I think the logic is easier to follow in the first function. Definitely biased, but I also like having the new custom function, and think it will be useful in the future based on how its set up (main reason I chose to do it that way - though could have made it a map call instead of a for-loop)

Reply:
> Im going to keep the functions the way they are for now,

Sure, Seth already approved it, so I think he agrees these are okay.  I'm just giving my opinion.

> as I think the logic is easier to follow in the first function.

I think using `map` on a vector is convoluted and you should just use more standard operations.  And the condition branching is unnecessary.  So, in my eyes, definitely not easier to follow.

I'd suggest at least move the conditions outside the map, because that's not changing within the loop.

Reply:
I had a feeling I should have waited to merge. Was in the mindset of 'checking off boxes' and it seemed good to go after the latest commit. I can make a quick new PR for that change if you'd like

Reply:
> I had a feeling I should have waited to merge

You merging is fine from my perspective.  I'm just trying to clarify my input/opinion, but the main reason I jumped in was to ask for [relocating the map_list_recursive](https://github.com/metrumresearchgroup/bbr/pull/524#discussion_r936137729) for compatibility with gh-527, and you addressed that (thanks).